src/adapters/http/client.ts

```diff
@@ -0,0 +1,72 @@
+import Adapter from "../../lib/adapter.js"
+import GleeMessage from "../../lib/message.js"
+import got from "got"
+import { HttpAdapterConfig } from "../../lib/index.js"
+import http from "http"
+class HttpClientAdapter extends Adapter {
+
+  name(): string {
+    return "HTTP client"
+  }
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  async _connect(): Promise<this> {
+    this.emit("connect", {
+      name: this.name(),
+      adapter: this,
+      connection: http,
+      channel: this.channelNames,
+    })
+    return this
+  }
+  async _send(message: GleeMessage): Promise<void> {
+    const headers = {}
+    const config: HttpAdapterConfig = await this.resolveProtocolConfig("http")
+    const clientConfig = config?.client
```

Comment:
This is not the optimal way to access the auth config now, check out https://github.com/asyncapi/glee/blob/8afea22cb3096b15f72f091670e27f48b79ab66a/src/adapters/ws/client.ts#L34